### PR TITLE
Showing confirmation dialog when user tries to delete an agent

### LINKED ISSUE #615: Show confirmation dialog when user tries to delete an agent.
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/homanp/superagent/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/homanp/superagent/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Show confirmation dialog when a user tries to delete an agent.

### 📝 Additional Context

_No response_

### Acknowledgements

- [X] My issue title is concise and descriptive.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

### PR DESCRIPTION
**Pull request by elisalimli**
## Summary

This PR fixes #615 

- [ ] My code follows the code style of this project and passes `make format`.
- [ ] My change requires a change to the documentation.
- [ ] I have [updated the documentation](https://docs.superagent.sh/) accordingly.
- [ ] My change has adequate unit test coverage.

### CODE REVIEW

**Comment by shuriken-bytes** [comment]
The code changes look good. It adds a new component called `DeleteAgentButton` that handles the deletion of an agent. It uses the `AlertDialog` component from `@/components/ui/alert-dialog` to display a confirmation dialog before deleting the agent. The `DeleteAgentButton` component is then used in the `header.tsx` file to replace the previous delete button implementation. Overall, the code changes seem to improve the user experience when deleting an agent. Great job, @elisalimli! 👍🎉

**Comment by RutamBhagat** [comment]
The dialog component can be more generic in order to be reusable across the frontend for increased code versatility

File: libs/ui/app/agents/[agentId]/header.tsx [hunk 1]
```diff
@@ -10,6 +10,17 @@ import { Agent } from "@/types/agent"
...
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from "@/components/ui/alert-dialog"
```
Comment (RutamBhagat) [inline libs/ui/app/agents/[agentId]/header.tsx:23]: Unused import statement

**Review by homanp — APPROVED** [review]
Looks good!

**Pull request merged**